src/main/java/complete_task/completeTaskUseCase.java

```diff
@@ -14,9 +16,14 @@ public class completeTaskUseCase implements completeTaskInputBoundary {
     /**
      * Complete the task by setting the Task's completion to 100.
      * @param taskId the unique id of the task to be completed.
+     * @param taskCurriculumId the unique id of the curriculum
+     *                         that the task belongs to.
+     * @param activeUser the active user using the program right now.
      */
     @Override
-    public void completeTask(int taskId, Curriculum taskCurriculum){
+    public void completeTask(int taskId, int taskCurriculumId, User activeUser){
+        Schedule schedule = activeUser.getSchedule();
```

Comment:
Instead of passing the active user as a parameter, you can access it directly here because of the singleton pattern, using InMemoryUser.getActiveUser();

Reply:
As far as I understand, this singleton will be passed as a parameter to all interaction instances, so that they never need update themselves, so to speak, in order to always access the currently logged-in user.